src/main/java/seedu/address/ui/CommandHelpMessage.java

```diff
@@ -0,0 +1,93 @@
+package seedu.address.ui;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.TitledPane;
+import seedu.address.logic.commands.AddCommand;
+import seedu.address.logic.commands.CommandEnum;
+import seedu.address.logic.commands.DeleteCommand;
+import seedu.address.logic.commands.EditCommand;
+
+public class CommandHelpMessage extends UiPart<TitledPane> {
+    private static final String FXML = "CommandHelpMessage.fxml";
+
+    // TODO: Shift help messages to respective command classes when implemented
+    private static final String FLIP_COMMAND_DESCRIPTION = "Toggles the flashcard to either show or hide its English"
+            + " translation.";
+    private static final String FLIP_COMMAND_USAGE = "flip INDEX";
+    private static final String FLIP_COMMAND_EXAMPLES = "flip 3";
+    private static final String TEST_COMMAND_DESCRIPTION = "Checks whether the English translation of a flashcard"
+            + " matches a given phrase.";
+    private static final String TEST_COMMAND_USAGE = "test INDEX a/ENGLISH_PHRASE";
+    private static final String TEST_COMMAND_EXAMPLES = "test 1 a/good morning";
+    private static final String IMPORT_COMMAND_DESCRIPTION = "Imports flashcards into LingoGO! from a specified"
+            + " CSV file";
+    private static final String IMPORT_COMMAND_USAGE = "upload CSV_FILE_PATH";
+    private static final String IMPORT_COMMAND_EXAMPLES = "upload ./dictionary.csv";
+    private static final String EXPORT_COMMAND_DESCRIPTION = "Exports flashcards to a CSV file located in the data"
+            + " folder";
+    private static final String EXPORT_COMMAND_USAGE = "download CSV_FILE_PATH";
+    private static final String EXPORT_COMMAND_EXAMPLES = "download";
+
+    @FXML
+    private TitledPane helpPane;
+    @FXML
+    private Label description;
+    @FXML
+    private Label usage;
+    @FXML
+    private Label examples;
+
+    private HelpWindow helpWindow;
+
+    /**
+     * Creates a new CommandHelpMessage for a specified command.
+     *
+     * @param command The command that the help message is explaining.
+     * @param helpWindow The HelpWindow that the CommandHelpMessage is contained in.
+     */
+    public CommandHelpMessage(CommandEnum command, HelpWindow helpWindow) {
+        super(FXML);
+        this.helpWindow = helpWindow;
+        this.getRoot().heightProperty().addListener((obs, oldHeight, newHeight) -> helpWindow.getRoot().sizeToScene());
+        switch (command) {
+        case ADD:
+            this.setDisplayText("Add", AddCommand.COMMAND_DESCRIPTION, AddCommand.COMMAND_USAGE,
+                    AddCommand.COMMAND_EXAMPLES);
```

Comment:
would be nice to wrap this in a method inside each Command

Reply:
It might break encapsulation as the method accesses the javfx controls directly to set their text, so the Command classes should not be responsible for showing the help messages?

Reply:
i meant doing something like `this.setDisplayText(AddCommand.getHelpText())`. this shouldn't break encapsulation? and adds a nice layer of abstraction over the help text (which should be changed inside the command class itself)